docs/Using Randomness/get-a-random-number.md

```diff
@@ -4,96 +4,257 @@ section: smartContract
+        LINKTOKEN = LinkTokenInterface(link);
+        s_requestConfig = RequestConfig({
+            subId: 0, // Unset
+            callbackGasLimit: callbackGasLimit,
+            requestConfirmations: requestConfirmations,
+            numWords: numWords,
+            keyHash: keyHash
+        });
     }
-    
-    /** 
-     * Requests randomness 
-     */
-    function getRandomNumber() public returns (bytes32 requestId) {
-        require(LINK.balanceOf(address(this)) >= fee, "Not enough LINK - fill contract with faucet");
-        return requestRandomness(keyHash, fee);
+
+    function fulfillRandomWords(
+        uint256 requestId,
+        uint256[] memory randomWords
+    )
+        internal
+        override
+    {
+        s_randomWords = randomWords;
     }
 
-    /**
-     * Callback function used by VRF Coordinator
-     */
-    function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
-        randomResult = randomness;
+    function requestRandomWords()
+        external
+    {
+        RequestConfig memory rc = s_requestConfig;
+        // Will revert if subscription is not set and funded.
+        s_requestId = COORDINATOR.requestRandomWords(
+            rc.keyHash,
+            rc.subId,
+            rc.requestConfirmations,
+            rc.callbackGasLimit,
+            rc.numWords);
     }
 
-    // function withdrawLink() external {} - Implement a withdraw function to avoid locking your LINK in the contract
+    // Assumes this contract owns link
+    function topUpSubscription(
+        uint256 amount
+    )
+        external
+    {
+        LINKTOKEN.transferAndCall(
+            address(COORDINATOR),
+            amount,
+            abi.encode(s_requestConfig.subId));
+    }
+
+    function unsubscribe()
+        external
+    {
+        // Returns funds to this address
+        COORDINATOR.cancelSubscription(s_requestConfig.subId, address(this));
+        s_requestConfig.subId = 0;
+    }
+
+    function subscribe()
+        external
+    {
+        address[] memory consumers = new address[](1);
+        consumers[0] = address(this);
+        s_requestConfig.subId = COORDINATOR.createSubscription(consumers);
+    }
+}
+``` 
+
+
+## Multiple consumers, external subscription owner
+
+In this example, the subscription for multiple consumers is managed by an external account. Steps to set up this configuration:
+
+1. Create a subscription with no consumers `createSubscription()`. Make a note of the subscriptionId emitted in the `SubscriptionCreated` log. TODO: metamask screen shots
+2. Deploy your applications which expose a `setSubscriptionId(uint64 subId)` like `VRFConsumerExternalSubOwner` does and record all their addresses. 
+3. Register all the applications `addConsumer(uint64 subId, address consumer)` . TODO: metamask screen shots
+4. Fund the subscription with `LINKTOKEN.transferAndCall(address(COORDINATOR), amount, abi.encode(subId));`
```

Comment:
```suggestion
1. Create a subscription with no consumers `createSubscription()`. Make a note of the subscriptionId emitted in the `SubscriptionCreated` log. TODO: metamask screen shots
1. Deploy your applications which expose a `setSubscriptionId(uint64 subId)` like `VRFConsumerExternalSubOwner` does and record all their addresses. 
1. Register all the applications `addConsumer(uint64 subId, address consumer)` . TODO: metamask screen shots
1. Fund the subscription with `LINKTOKEN.transferAndCall(address(COORDINATOR), amount, abi.encode(subId));`
```

Reply:
why all 1s here - accidental?

Reply:
Markdown handles the ordering automatically. Setting them all to 1. makes it easier to re-order them in the future or insert additional items to the middle of the list. https://rdmd.readme.io/docs/lists#auto-ordering